app/services/users/syncrhonize_removed_users_service.rb

```diff
@@ -0,0 +1,17 @@
+module Users
+  class SyncrhonizeRemovedUsersService
+    attr_reader :users
+
+    def initialize(users)
+      @users = users
+    end
+
+    def perform
+      logins = @users.map{ |user| user.login }
+      User.where.not(login: logins).each do |user|
+        user.active = false
+        user.save!
```

Comment:
are we good here running validations with `save!`, or can that be avoided ?, if so, you could write these lines like `User.where.not(login: logins).update_all(active: false)`. But this is only if we can avoid running validations.

Reply:
Awesome, thank you so much for that suggestion, I didn't know about that method 🤓 👍